src/stirling/source_connectors/socket_tracer/openssl_trace_bpf_test.cc

```diff
@@ -151,9 +167,22 @@ http::Record GetExpectedHTTPRecord() {
 typedef ::testing::Types<NginxOpenSSL_1_1_0_ContainerWrapper, NginxOpenSSL_1_1_1_ContainerWrapper,
                          Node12_3_1ContainerWrapper, Node14_18_1AlpineContainerWrapper>
     OpenSSLServerImplementations;
-TYPED_TEST_SUITE(OpenSSLTraceTest, OpenSSLServerImplementations);
 
-TYPED_TEST(OpenSSLTraceTest, ssl_capture_curl_client) {
+template <typename T>
+using OpenSSLTraceDlsymTest = BaseOpenSSLTraceTest<T, false>;
+
+template <typename T>
+using OpenSSLTraceRawFptrsTest = BaseOpenSSLTraceTest<T, true>;
+
+#define OPENSSL_TYPED_TEST(TestCase, CodeBlock)            \
+  TYPED_TEST(OpenSSLTraceDlsymTest, TestCase)              \
+  CodeBlock TYPED_TEST(OpenSSLTraceRawFptrsTest, TestCase) \
+  CodeBlock
+
+TYPED_TEST_SUITE(OpenSSLTraceDlsymTest, OpenSSLServerImplementations);
+TYPED_TEST_SUITE(OpenSSLTraceRawFptrsTest, OpenSSLServerImplementations);
+
+OPENSSL_TYPED_TEST(ssl_capture_curl_client, {
```

Comment:
Clever. Though also wonder if there's a better way. Apparently, we can't mix typed parameters with value parameters in gtest, which is unfortunate. So this might be the best way.

In theory, you might be able to turn the second value as a second type parameters, such that we `OpenSSLServerImplementations` gets converted to a sequence of `std::pairs`(like https://stackoverflow.com/questions/52256953/c-multiple-parameters-with-gtest-typed-test).

You can feel free to leave this as is, though, since it may not be worth the effort to change this.

Reply:
I tried something similar to that, but ran into difficulties so I'd prefer to leave this as is.